runtime/src/snapshot_utils.rs

```diff
@@ -1120,7 +1120,15 @@ fn archive_snapshot(
                 .append_dir_all(SNAPSHOTS_DIR, &staging_snapshots_dir)
                 .map_err(E::ArchiveSnapshotsDir)?;
 
-            for storage in snapshot_storages {
+            let mut sorted_storage_indices = (0..snapshot_storages.len()).collect::<Vec<_>>();
+            sorted_storage_indices.sort_by_key(|&i| snapshot_storages[i].accounts.len());
+            for i in 0..sorted_storage_indices.len() {
+                // Balance large and small files with bias towards small (4 small + 1 large), such
+                // that during unpacking large writes are mixed with file metadata operations
+                // and towards the end of archive (sizes equalize) writes are >256KiB / file.
+                let indices_range = 0..sorted_storage_indices.len();
+                let index = select_from_range_with_start_end_rates(indices_range, i, 4, 1);
```

Comment:
nit: make the magic number 4, 1 into constant with meaningful variable names?

Reply:
I will do a follow-up PR to change this, since it includes a slight refactor that touches most of the diff...